### PR TITLE
GEODE-6954 Update dispatcher in JMX after receiver restart

### PR DESCRIPTION
**Pull request by mkevo**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [X] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [X] Is your initial contribution a single, squashed commit?

- [X] Does `gradlew build` run cleanly?

- [X] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/beans/GatewaySenderMBeanBridge.java [hunk 1]
```diff
@@ -293,9 +297,29 @@ public String getGatewayReceiver() {
   public boolean isConnected() {
     if (this.dispatcher != null && this.dispatcher.isConnectedToRemote()) {
       return true;
+    }
+    if (!this.sender.isParallel()) {
```
Comment (jmelchio) [inline geode-core/src/main/java/org/apache/geode/management/internal/beans/GatewaySenderMBeanBridge.java]: Would prefer to see this for easier readability: 

```java
if(this.sender.isParallel)) { 
  // code for parallel
} else {
  // code for serial
}
```

File: geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ConcurrentParallelGatewaySenderEventProcessor.java [hunk 2]
```diff
@@ -309,9 +309,6 @@ protected void waitForResumption() throws InterruptedException {
     super.waitForResumption();
   }
 
-  /**
-   * Test only methods for verification purpose.
-   */
   public List<ParallelGatewaySenderEventProcessor> getProcessors() {
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ConcurrentParallelGatewaySenderEventProcessor.java]: It's ok to remove the javadocs, but you should add the annotation `@VisibleForTesting`.

Any access modifier (private/package-private/protected/public) that is made less restrictive than `private` simply to allow a test to access it should have `@VisibleForTesting` added to it.

Comment (mkevo) [reply geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ConcurrentParallelGatewaySenderEventProcessor.java]: I remove this javadoc as we don't use this method only for testing.

**Comment by mkevo** [comment]
@kirklund, @jmelchio,  thanks for the comments! 
New commit is ready.

**Pull request merged**